### PR TITLE
The scheduler supports the scheduling algorithm of the map-reduce str…

### PR DESCRIPTION
**Pull request by dingtsh1**
The scheduler supports the map-reduce structure algorithm. We can use the plugin to register the NodeReduceFn to reduce the NodeMapFn calculation result.

### CODE REVIEW

File: pkg/scheduler/framework/session_plugins.go [hunk 1]
```diff
@@ -65,6 +66,11 @@ func (ssn *Session) AddNodeOrderFn(name string, pf api.NodeOrderFn) {
 	ssn.nodeOrderFns[name] = pf
 }
 
+// AddNodeReduceFn add Node reduce function
+func (ssn *Session) AddNodeReduceFn(name string, pf api.NodeReduceFn) {
```
Comment (k82cn) [inline pkg/scheduler/framework/session_plugins.go:75]: Let's add `AddNodeMapFn` to work together with this reduce function.

Comment (dingtsh1) [reply pkg/scheduler/framework/session_plugins.go:75]: Done

**Review by k82cn — APPROVED** [review]

**Pull request merged**